### PR TITLE
feat(flags): add relative date operators and fix numeric ops

### PR DESCRIPTION
**Pull request by liyiy**
ruby version of the relative date operator add and numeric ops fixes for local evaluation

https://github.com/PostHog/posthog-python/pull/105

### CODE REVIEW

File: spec/posthog/feature_flag_spec.rb [hunk 1]
```diff
@@ -1220,6 +1234,121 @@ class PostHog
...
+        expect(FeatureFlagsPoller.match_property(property_n, { 'key' => '2022-05-01 00:00:00' })).to be true
+        expect(FeatureFlagsPoller.match_property(property_n, { 'key' => '2021-05-01 00:00:01' })).to be true
+        expect(FeatureFlagsPoller.match_property(property_n, { 'key' => '2021-05-01 00:00:00' })).to be false
+        expect(FeatureFlagsPoller.match_property(property_n, { 'key' => '2021-04-30 00:00:00' })).to be false
+        expect(FeatureFlagsPoller.match_property(property_n, { 'key' => '2021-03-01 12:13:00' })).to be false
+
+      end
+    end
+
+    it 'nil property value with all operators' do
```
Comment (liyiy) [inline spec/posthog/feature_flag_spec.rb:1339]: Do we need to test this fully for ruby? Not sure it makes sense here since nil just returns empty when it's stringified. Unless we want to force nil values to be a string of nil.

Comment (neilkakkar) [reply spec/posthog/feature_flag_spec.rb:1339]: nope, makes sense, just make sure that operations don't crash with nil where we don't expect them to

File: spec/posthog/feature_flag_spec.rb [hunk 2]
```diff
@@ -1336,19 +1336,169 @@ class PostHog
       end
     end
 
-    it 'nil property value with all operators' do
+    # def test_none_property_value_with_all_operators(self):
```
Comment (liyiy) [inline spec/posthog/feature_flag_spec.rb:1339]: some leftover copy and paste (ignore if you're still WIPing it)

Comment (neilkakkar) [reply spec/posthog/feature_flag_spec.rb:1339]: nah this one's ready, will remove thanks!
